File: data/transcoder_evaluation_gfg/go/PROGRAM_CIRCUMFERENCE_PARALLELOGRAM.go
package main

import (
	"math"
		"fmt"
	"os"
	"unsafe"
)

func min(x int, y int) int {
	if x < y {
		return x
	}
	return y
}
func max(x int, y int) int {
	if x > y {
		return x
	}
	return y
}
func cmpfunc(a unsafe.Pointer, b unsafe.Pointer) int {
	return *(*int)(a) - *(*int)(b)
}


func f_gold(a float32, b float32) float32 {
	return (a * 2) + b*2
}
//TOFILL
func main() {
	var (
		n_success int         = 0
		param0    []float32 = []float32{801.0366882228715, -7069.610056819919, 7723.966966568705, -7935.859205856963, 6094.247432557289, -7371.490363309265, 8368.473889617526, -3761.921143166053, 3139.1089185587884, -5218.286665567171}
		param1    []float32 = []float32{456.71190645582783, -4226.483870778477, 5894.65405158763, -5333.225064296693, 1660.420120702062, -1095.4543576847332, 4735.838330834498, -5315.871691690649, 6490.194159517967, -8265.153014320813}
	)
	for i := int(0); i < len(param0[:]); i++ {
		if float64(math.Abs(float64(1-(float64(math.Abs(float64(f_gold(param0[i], param1[i]))))+1e-07)/(float64(math.Abs(float64(f_filled(param0[i], param1[i]))))+1e-07)))) < 0.001 {
			n_success += 1
		}
	}
	fmt.Print("#Results:", " ", n_success, ", ", len(param0[:]))
	os.Exit(0)
}

File: data/transcoder_evaluation_gfg/c/PROGRAM_CIRCUMFERENCE_PARALLELOGRAM.c
// Copyright (c) 2019-present, Facebook, Inc.
// All rights reserved.
//
// This source code is licensed under the license found in the
// LICENSE file in the root directory of this source tree.
//


#include <stdio.h>
#include <math.h>
#include <stdlib.h>
#include <limits.h>
#include <stdbool.h>

int min(int x, int y) { return (x < y)? x: y; }
int max(int x, int y) { return (x > y)? x: y; }
int cmpfunc (const void * a, const void * b) {return ( *(int*)a - *(int*)b );}
int len (int arr [ ]) {return ((int) (sizeof (arr) / sizeof (arr)[0]));}
void sort (int arr [ ], int n) {qsort (arr, n, sizeof(int), cmpfunc);}

float f_gold ( float a, float b ) {
  return ( ( 2 * a ) + ( 2 * b ) );
}


float f_filled ( float a, float b ) {}

int main(void) {
    int n_success = 0;
    float param0[] = {801.0366882228715F,-7069.610056819919F,7723.966966568705F,-7935.859205856963F,6094.247432557289F,-7371.490363309265F,8368.473889617526F,-3761.921143166053F,3139.1089185587884F,-5218.286665567171F};
    float param1[] = {456.71190645582783F,-4226.483870778477F,5894.65405158763F,-5333.225064296693F,1660.420120702062F,-1095.4543576847332F,4735.838330834498F,-5315.871691690649F,6490.194159517967F,-8265.153014320813F};
    for(int i = 0; i < len(param0); ++i)
    {
        if(abs(1 - (0.0000001 + abs(f_gold(param0[i],param1[i])) )/ (abs(f_filled(param0[i],param1[i])) + 0.0000001)) < 0.001F)
        {
            n_success+=1;
        }
    }
    printf("#Results:", " ", n_success, ", ", len(param0));
    return 0;
}